README.md
# sketch-reference-files

> A store of automatically generated Sketch file JSON organised by document version and Sketch feature.

## Overview

Sketch documents are stored as ZIP archives containing JSON encoded data. The file format was
originally introduced in Sketch 43 and allows for better third-party integration. For example
generating, reading and modifying documents without opening them in Sketch.

In order to provide concrete examples of how different Sketch features are serialised into JSON this
repo maintains an archive of reference files.

It's published as an npm module to aid use cases that may involve importing these reference files as
test fixtures.

## Related projects

- [sketch-file-format](https://github.com/sketch-hq/sketch-file-format)

## Usage

### JavaScript

Add the npm module using `npm` or `yarn`

```sh
npm install @sketch-hq/sketch-reference-files
```

```js
import files from '@sketch-hq/sketch-reference-files'
```

The module exports reference file metadata and json contents for each supported document `version`.

```
[
  {
    document: 121,
    sketchVersions: ['59', '60', '61'],
    files: [
      { id, name, description, data },
      ...
    ],
  },
  ...
]
```

### HTTP

The raw reference file JSON can be accessed over HTTP via predicable urls:

- https://unpkg.com/browse/@sketch-hq/sketch-reference-files@latest/files/

## Development

This section of the readme is related to developing the reference file repo. If you just want to consume the reference file data you can safely ignore this.

### Requirements

- Yarn >= `1.13`
- Node `12.*.*`

### Scripts

| Script              | Description                                                                                                                                                                                                                                       |
| ------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| yarn download-apps  | Download and locally cache all Sketch app binaries needed to generate the files                                                                                                                                                                   |
| yarn generate-files | Generate reference files into the `files/` folder. Existing reference files are skipped, so delete the `files/` folder if you need to generate everything from scratch. This command needs to be run manually, and isn't part of the normal build |
| yarn build          | Builds the module into the the `dist/` folder                                                                                                                                                                                                     |
| yarn changeset      | Open an interactive CLI to add a changeset                                                                                                                                                                                                        |
| yarn commit         | Open an interactive CLI to make a git commit in the conventional commits format                                                                                                                                                                   |

### Making changes

#### Adding reference files for a new Sketch version

1. Update the `scripts/config.ts` file with metadata for the new Sketch version
1. Run `yarn download-apps` to ensure you have the correct Sketch binaries available locally
1. Run `yarn generate-files` to build the reference files
1. PR your changes

#### Add a new reference file feature type

1. Add plugin code to demonstrate the feature in `features/`. Use the other features as a guide
1. Update the `scripts/config.ts` file with metadata for the file feature
1. Run `yarn download-apps` to ensure you have the correct Sketch binaries available locally
1. Run `yarn generate-files` to build the reference files
1. PR your changes

#### Conventional commits

Try and use the [conventional commits](https://www.conventionalcommits.org/) convention when writing commit messages. This isn't enforced, but you can use the `yarn commit` command (in place of `git commit -m "foo"`) to open an interactive CLI to walk you through generating a properly formatted commit message.

#### Releases

This repo uses [Atlassian Changesets](https://github.com/atlassian/changesets) to automate the npm release process. Read the docs for more information, but the top-level summary is:

- A GitHub Action maintains a permanently open PR that when merged will publish the package to npm with the latest changes and an automatically determined semver
- If the work you do in a PR should affect the next release, then you need to commit a "changeset" to the repo together with the rest of your code changes - do this by running `yarn changeset`. You'll be asked to provide a change type (major, minor or patch) and a message

## Browse

> ⚠️ This section is automatically generated. Any manual edits will be erased during a build.

### Document 123

> Sketch versions: 62

| Feature          | Document                                                                                                       | Pages                                                                                                  | Meta                                                                                                       | User                                                                                                       |
| ---------------- | -------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- |
| Empty            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/empty/document.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/empty/pages)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/empty/meta.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/empty/user.json)            |
| Groups           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/groups/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/groups/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/groups/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/groups/user.json)           |
| Images           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/images/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/images/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/images/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/images/user.json)           |
| Library styles   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-styles/document.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-styles/pages)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-styles/meta.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-styles/user.json)   |
| Library symbols  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-symbols/document.json)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-symbols/pages)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-symbols/meta.json)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/library-symbols/user.json)  |
| Prototypes       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/prototypes/document.json)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/prototypes/pages)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/prototypes/meta.json)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/prototypes/user.json)       |
| Shape paths      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shape-paths/document.json)      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shape-paths/pages)      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shape-paths/meta.json)      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shape-paths/user.json)      |
| Shapes           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shapes/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shapes/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shapes/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/shapes/user.json)           |
| Symbol overrides | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbol-overrides/document.json) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbol-overrides/pages) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbol-overrides/meta.json) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbol-overrides/user.json) |
| Symbols          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbols/document.json)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbols/pages)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbols/meta.json)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/symbols/user.json)          |
| Text             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/text/document.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/text/pages)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/text/meta.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/text/user.json)             |
| Smart layout     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/smart-layout/document.json)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/smart-layout/pages)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/smart-layout/meta.json)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/smart-layout/user.json)     |
| Blur             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/blur/document.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/blur/pages)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/blur/meta.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/blur/user.json)             |
| Slice            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/slice/document.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/slice/pages)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/slice/meta.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/slice/user.json)            |
| Export formats   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/export-formats/document.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/export-formats/pages)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/export-formats/meta.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/export-formats/user.json)   |
| Variable font    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/variable-font/document.json)    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/variable-font/pages)    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/variable-font/meta.json)    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/123/variable-font/user.json)    |

### Document 121

> Sketch versions: 59, 60, 61

| Feature          | Document                                                                                                       | Pages                                                                                                  | Meta                                                                                                       | User                                                                                                       |
| ---------------- | -------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- |
| Empty            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/empty/document.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/empty/pages)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/empty/meta.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/empty/user.json)            |
| Groups           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/groups/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/groups/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/groups/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/groups/user.json)           |
| Images           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/images/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/images/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/images/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/images/user.json)           |
| Library styles   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-styles/document.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-styles/pages)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-styles/meta.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-styles/user.json)   |
| Library symbols  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-symbols/document.json)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-symbols/pages)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-symbols/meta.json)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/library-symbols/user.json)  |
| Prototypes       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/prototypes/document.json)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/prototypes/pages)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/prototypes/meta.json)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/prototypes/user.json)       |
| Shape paths      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shape-paths/document.json)      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shape-paths/pages)      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shape-paths/meta.json)      | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shape-paths/user.json)      |
| Shapes           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shapes/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shapes/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shapes/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/shapes/user.json)           |
| Symbol overrides | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbol-overrides/document.json) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbol-overrides/pages) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbol-overrides/meta.json) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbol-overrides/user.json) |
| Symbols          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbols/document.json)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbols/pages)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbols/meta.json)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/symbols/user.json)          |
| Text             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/text/document.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/text/pages)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/text/meta.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/text/user.json)             |
| Smart layout     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/smart-layout/document.json)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/smart-layout/pages)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/smart-layout/meta.json)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/smart-layout/user.json)     |
| Blur             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/blur/document.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/blur/pages)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/blur/meta.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/blur/user.json)             |
| Slice            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/slice/document.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/slice/pages)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/slice/meta.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/slice/user.json)            |
| Export formats   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/export-formats/document.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/export-formats/pages)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/export-formats/meta.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/export-formats/user.json)   |
| Variable font    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/variable-font/document.json)    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/variable-font/pages)    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/variable-font/meta.json)    | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/121/variable-font/user.json)    |

### Document 120

> Sketch versions: 58

| Feature          | Document                                                                                                       | Pages                                                                                                  | Meta                                                                                                       | User                                                                                                       |
| ---------------- | -------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- |
| Empty            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/empty/document.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/empty/pages)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/empty/meta.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/empty/user.json)            |
| Groups           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/groups/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/groups/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/groups/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/groups/user.json)           |
| Images           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/images/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/images/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/images/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/images/user.json)           |
| Library styles   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-styles/document.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-styles/pages)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-styles/meta.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-styles/user.json)   |
| Library symbols  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-symbols/document.json)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-symbols/pages)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-symbols/meta.json)  | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/library-symbols/user.json)  |
| Prototypes       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/prototypes/document.json)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/prototypes/pages)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/prototypes/meta.json)       | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/prototypes/user.json)       |
| Shape paths      | -                                                                                                              | -                                                                                                      | -                                                                                                          | -                                                                                                          |
| Shapes           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/shapes/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/shapes/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/shapes/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/shapes/user.json)           |
| Symbol overrides | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbol-overrides/document.json) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbol-overrides/pages) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbol-overrides/meta.json) | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbol-overrides/user.json) |
| Symbols          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbols/document.json)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbols/pages)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbols/meta.json)          | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/symbols/user.json)          |
| Text             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/text/document.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/text/pages)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/text/meta.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/text/user.json)             |
| Smart layout     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/smart-layout/document.json)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/smart-layout/pages)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/smart-layout/meta.json)     | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/smart-layout/user.json)     |
| Blur             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/blur/document.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/blur/pages)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/blur/meta.json)             | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/blur/user.json)             |
| Slice            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/slice/document.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/slice/pages)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/slice/meta.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/slice/user.json)            |
| Export formats   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/export-formats/document.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/export-formats/pages)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/export-formats/meta.json)   | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/120/export-formats/user.json)   |
| Variable font    | -                                                                                                              | -                                                                                                      | -                                                                                                          | -                                                                                                          |

### Document 119

> Sketch versions: 55.2, 56, 56.1, 56.2, 56.3, 57, 57.1

| Feature          | Document                                                                                                       | Pages                                                                                                  | Meta                                                                                                       | User                                                                                                       |
| ---------------- | -------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------- |
| Empty            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/empty/document.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/empty/pages)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/empty/meta.json)            | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/empty/user.json)            |
| Groups           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/groups/document.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/groups/pages)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/groups/meta.json)           | [🔗](https://github.com/sketch-hq/sketch-reference-files/blob/master/files/119/groups/user.json)           |

CHANGELOG.md
# @sketch-hq/sketch-reference-files

## 2.3.0
### Minor Changes

- a2985a8: generate files with sketch 63

### Patch Changes

- 584ac2d: do not run changeset status on circle
- 86c8db0: setting up changesets
- 86c8db0: remove conventional commit linting


scripts/config.ts
import { Range } from 'semver'

export type Feature = {
  id: string // Feature file name/id
  range: Range // Compatible Sketch versions
  name: string // Human readable name
  description: string // Human readable description
}

export const features: Feature[] = [
  {
    id: 'empty',
    range: new Range('*'),
    name: 'Empty',
    description:
      'This is just a bare-bones, empty Sketch document. The equivalent of creating a new document and saving it immediately.',
  },
  {
    id: 'groups',
    range: new Range('*'),
    name: 'Groups',
    description: 'This document demonstrates nested group and layer hierarchy.',
  },
  {
    id: 'images',
    range: new Range('*'),
    name: 'Images',
    description: 'This document contains a bitmap image layer.',
  },
  {
    id: 'library-styles',
    range: new Range('*'),
    name: 'Library styles',
    description:
      'This document contains a shared style imported from a library and applied to a shape on the document',
  },
  {
    id: 'library-symbols',
    range: new Range('*'),
    name: 'Library symbols',
    description: 'This document contains a symbol imported from a library and placed on the page',
  },
  {
    id: 'prototypes',
    range: new Range('*'),
    name: 'Prototypes',
    description: 'This document demonstrates prototypes by creating a link between two artboards.',
  },
  {
    id: 'shape-paths',
    range: new Range('<58.0.0 || >=59.0.0'), // 58 has a bug that means this feature's plugin breaks
    name: 'Shape paths',
    description: 'This document contains a vector shape path layer.',
  },
  {
    id: 'shapes',
    range: new Range('*'),
    name: 'Shapes',
    description: 'This document contains a single styled shape.',
  },
  {
    id: 'symbol-overrides',
    range: new Range('*'),
    name: 'Symbol overrides',
    description: 'This document demonstrates overriding text values in symbols.',
  },
  {
    id: 'symbols',
    range: new Range('*'),
    name: 'Symbols',
    description:
      'This document contains a symbol master and one of its instances, side-by-side on the same page.',
  },
  {
    id: 'text',
    range: new Range('*'),
    name: 'Text',
    description: 'This document contains styled text layers.',
  },
  {
    id: 'smart-layout',
    range: new Range('>=58.0.0'), // Smart layout only available from Sketch 58
    name: 'Smart layout',
    description: 'This document demonstrates symbols using smart layouts.',
  },
  {
    id: 'blur',
    range: new Range('*'),
    name: 'Blur',
    description: 'Applies a blur to a layer',
  },
  {
    id: 'slice',
    range: new Range('*'),
    name: 'Slice',
    description: 'Creates a slice layer',
  },
  {
    id: 'export-formats',
    range: new Range('*'),
    name: 'Export formats',
    description: 'Applies export formats to an artboard',
  },
  {
    id: 'variable-font',
    range: new Range('>=59.0.0'), // Variable fonts only available from Sketch 59
    name: 'Variable font',
    description: 'Sets values on a variable font',
  },
]

export type VersionInfo = {
  document: number // Document version
  sketchVersions: [string, string][] // Tuple [sketchVersion, sketchBuild]
}

export const versions: VersionInfo[] = [
  {
    document: 118,
    sketchVersions: [['55', '78076'], ['55.1', '78136']],
  },
  {
    document: 119,
    sketchVersions: [
      ['55.2', '78181'],
      ['56', '81588'],
      ['56.1', '81669'],
      ['56.2', '81672'],
      ['56.3', '81716'],
      ['57', '83077'],
      ['57.1', '83088'],
    ],
  },
  {
    document: 120,
    sketchVersions: [['58', '84663']],
  },
  {
    document: 121,
    sketchVersions: [['59', '86127'], ['60', '88103'], ['61', '89581']],
  },
  {
    document: 123,
    sketchVersions: [['62', '91390'], ['63', '92445'], ['63.1', '92452']],
  },
]
